### ISSUE TITLE
PHP 8.2RC2 bug with first page load after deploy when no cache/container compiled yet. 

### ISSUE THREAD

**Issue by PhilETaylor** [symfony/symfony#47653]
### Symfony version(s) affected

6.1.4 

### Description

Ok I appreciate this is a piss-poor bug report but it's all I have at the moment. Im logging it here in case others see the same thing, and if those with better Symfony skills have a 💡 moment and know exactly what the problem is.

### How to reproduce

Using PHP 8.2RC ... (I know I know) 

When deployed, on the very very first page load (so probably when compiling and caching the container) Im getting one of a few exceptions thrown (and caught into Sentry.io) 

On the second page load, and on every single page load **after the first page load the whole app is exception free and working fine, including the lines in the code that raise this initial exception - it all works perfectly in PHP 8.2 - once that first page load/cache/container compile is complete.** 

It also did not raise any exceptions using PHP 7/8/8.1 - this part of my codebase is untouched throughout the PHP cycles. With the exception of adding `readonly` to the constructor property I guess. 

Their exception messages are truncated and incomplete - but are the full message. 

`"Cannot instantiate interface from"`

and sometimes 

`"Error: Cannot instantiate trait from"`

https://sentry.io/share/issue/de89a5060617453f9e660714e5cb5b9e/

```
Error: Cannot instantiate interface from
#6 /src/MyJoomla/Controller/UpdatesController.php(615): MyJoomla\Controller\UpdatesController::doExtensionUpgrade
#5 /vendor/symfony/http-kernel/HttpKernel.php(153): Symfony\Component\HttpKernel\HttpKernel::handleRaw
#4 /vendor/symfony/http-kernel/HttpKernel.php(75): Symfony\Component\HttpKernel\HttpKernel::handle
#3 /vendor/symfony/http-kernel/Kernel.php(202): Symfony\Component\HttpKernel\Kernel::handle
#2 /vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php(35): Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner::run
#1 /vendor/autoload_runtime.php(29): require_once
#0 /public/index.php(15): null
```

Now, when inspecting my own code, I see that Im instantiating a Message and then dispatching it in the bus. 

<img width="449" alt="Screenshot 2022-09-21 at 16 29 29" src="https://user-images.githubusercontent.com/400092/191546713-928fb313-6e98-4b3b-a1c5-6775e3a55def.png">

My message class is simple: 

```php
<?php
namespace App\Message;

class RemoteLogMessage extends AbstractAsyncMessage
{
}
```
which extends the abstract class 

```php
<?php
namespace App\Message;

abstract class AbstractAsyncMessage
{
    public function __construct(
        private readonly array $data = []
    ) {
    }

    public function getData(): array
    {
        return $this->data;
    }
}
```

### Possible Solution

No idea what the root cause or solution is. 

One guess is that an abstract class *must* need one or more abstract methods? or not? its valid without else it would not work?

This error cannot be replicated on demand - but it has happened the last few times I have deployed to production - within seconds. (Production is a docker container in a docker swarm service)

### Additional Context

Please no lectures on running PHP 8.2 RC2 in production ... someone has to find the bugs in the real world right :) hehe. 

First :) 
<img width="615" alt="Screenshot 2022-09-21 at 16 37 14" src="https://user-images.githubusercontent.com/400092/191548469-261b787d-9631-4f75-9941-1fa712aa4484.png">

**Comment by PhilETaylor** [comment]
Ok so digging in the PHP source I get to 

https://github.com/php/php-src/blob/adb45a63c0f56bb4e04ffd74579de0c8ab9d722f/Zend/zend_API.c#L1669

Which is preceded with 

```
/* This function requires 'properties' to contain all props declared in the
 * class and all props being public. If only a subset is given or the class
 * has protected members then you need to merge the properties separately by
 * calling zend_merge_properties(). */
```

But that doesn't explain why Symfony only errors on the first page load and not thereafter ... so Im unsure if Im hitting a PHP 8.2 bug or a symfony bug...

**Comment by PhilETaylor** [comment]
ok dug out another example - this time is nothing to do with abstract classes, its a Mail message being sent

<img width="1052" alt="Screenshot 2022-09-21 at 16 51 19" src="https://user-images.githubusercontent.com/400092/191551671-0f6a965e-a302-4f7b-a692-839733b467eb.png">

**Comment by stof** [comment]
As you are definitely not instantiating an interface here but a concrete class, I would qualify that as a PHP bug and report it to them instead.

**Comment by PhilETaylor** [comment]
Our posts crossed. As you can see from my second example, there I am simply passing MailerInterface my Email to send.. which are both Symfony core classes used correctly - which rules out my own message and abstract class. 

Im leaning now to this being a PHP bug not a symfony bug too after seeing this example. I just wish I could reproduce it on demand before reporting it :(

**Comment by stof** [comment]
Well, in the second case, the bug happens on `new BlockTokenParser`, which is also an instantiation (but for a concrete class)

**Comment by PhilETaylor** [comment]
Just a note to say this is still happening in 8.2RC3 :)

**Comment by mfn** [comment]
Apologies @PhilETaylor , has there been a bug report created for PHP itself? I can't see a link anywhere.

**Comment by PhilETaylor** [comment]
I have not been able to replicate it on demand or reproduce it in testing, and so I have not reported it out of fear that it's just me.

It only happens once, after first deploying to production, and not instantly, about 2-5mins after starting the container, and the class and location in the code seems random

No one else seems to have this issue - yet, and the PHP source code reporting the error message has not been touched for years.

**Comment by PhilETaylor** [comment]
Bug logged at php-src https://github.com/php/php-src/issues/9675

**Comment by PhilETaylor** [comment]
This seems resolved in PHP 8.2 RC 6 released this week so "maybe" connected to something contained in https://news-web.php.net/php.internals/118991

**Comment by PhilETaylor** [comment]
This has been confirmed as an issue by others now, I wasn't going mad haha... Bug logged at php-src https://github.com/php/php-src/issues/9675
